### ISSUE TITLE
Github tool - Not able to get comments for Issues/PRs for private github repos. Also not able to list private repos .

### ISSUE THREAD

**Issue by sangee2004** [obot-platform/obot#174]
Using our github tool, I am able to list issues but not list comments for an issue for `http://github.com/otto8-ai/otto8/` repo. I get this error . 

```
I am unable to access the comments for this issue due to OAuth App access restrictions set by the otto8-ai organization. You may need to view the comments directly on GitHub or adjust the repository's access settings.
```

Same behavior is also seen with PRs. Able to list them, but not get details/comments.
`List repos` also does not list private repos.

**Note** - This issue is seen only for this repo which is private. I am able to get  comments for Issues/PRs from other public repos.

**Comment by sangee2004** [comment]
I am able to work with private repo that I own . Able to get and add comments to issues and PRs in the private repo.

I am not able to see the private repo being returned by `List repos` call

<img width="706" alt="Image" src="https://github.com/user-attachments/assets/c64276ff-f5ad-43db-8f11-2f24e7b7a623" />



@drpebcak Should we expect to see the private repos that user has access to be listed as well in this case?

**Comment by drpebcak** [comment]
No.. the response given here is accurate. `Please ensure that the repository "testrepo" exists and is accessible.`

**Comment by drpebcak** [comment]
There is nothing else to change in the tool, this message is expected when you have not granted the appropriate permissions in github.
